src/SerializerGenerator.php

```diff
@@ -184,18 +184,7 @@ private function generateCodeForFieldType(
         string $modelPropertyPath,
         array $stack
     ): string {
-        if ($type instanceof PropertyTypeArray) {
-            if ($type->getSubType() instanceof PropertyTypePrimitive) {
-                // for arrays of scalars, copy the field even when its an empty array
-                return $this->templating->renderArrayAssign($fieldPath, $modelPropertyPath);
```

Comment:
should we recursively dig into the sub type of the array to find if the innermost type is primitive or not?

if i understand correctly, your fix will generate code that loops over the outer arrays and only directly assigns the innermost array. but we could assign any depth of array of primitives directly i think.

Reply:
I just refactored this part to work more general so that this if statement is not needed anymore. One more method call to render the array does not hurt imo.